scrapy/extensions/robotstxtparser.py

```diff
@@ -0,0 +1,52 @@
+from six.moves.urllib_robotparser import RobotFileParser
+
+from scrapy.utils.python import to_native_str
+
+class BaseRobotsTxtParser():
```

Comment:
Maybe this class should not exist in code, but only in the documentation. I think this is the usual approach in Scrapy.